docs/kubernetes/mailu/front.yaml

```diff
@@ -1,23 +1,41 @@
-
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: apps/v1beta2
+kind: DaemonSet
 metadata:
   name: mailu-front
   namespace: mailu-mailserver
+  labels:
+    k8s-app: mail-loadbalancer
+    component: ingress-controller
+    type: nginx
 spec:
-  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: mail-loadbalancer
+      component: ingress-controller
+      type: nginx
   template:
     metadata:
       labels:
-        app: mailu-front
-        role: mail
-        tier: backend
+        k8s-app: mail-loadbalancer
+        component: ingress-controller
+        type: nginx
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-role.kubernetes.io/master
+                operator: DoesNotExist
```

Comment:
My current main deployment has master taint turned off, specifically it's a two-node system where both nodes are masters. This affinity selection would mean it runs on no nodes.

Even if that were not true, I'm not sure I want the containers running on every worker node.

Reply:
> Even if that were not true, I'm not sure I want the containers running on every worker node.

Of course it's to the end user to decide whether he wants it or not. But to enable HA I think this is the most easy solution. 
The whole `nodeSelectorTerms` can of course be adapted to only apply to a few nodes only. This can be easily adapted by a user.

> My current main deployment has master taint turned off, specifically it's a two-node system where both nodes are masters. This affinity selection would mean it runs on no nodes.

This is true but can be easily changed to (If I remember correctly):
```
- matchExpressions: 
  - key: node-role.kubernetes.io/worker
     operator: Exists
```

> Ideally, the nodes would be autoscaling rather than pinned to specific nodes.

 I think you mean the pods would be autoscaling, but this is mentioned above and gives you no real certainty about which nodes will run the `Deployment`

Reply:
Correct, apologies. I couldn't even see this response until digging around, I thought I didn't send it at all! :/

In any event, that does help, it puts the pod on every node. I'll experiment with setting up a HAProxy load balancer. I don't want to interrupt your flow with features you aren't interested in.